### PR TITLE
Feat/add events to docs

### PR DESCRIPTION
**Pull request by brendan-defi**
**What changed? Why?**
1. Created a `logEvent` utility in `libs` for use in `base-docs`. This mirrors the utility currently in `web`.
2. Created custom components for navbarLinks and dropdownLinks. These components integrate `logEvent`.
3. Changed the `type` of navbar components to the custom types as necessary to capture the events we want to track. 

**Notes to reviewers**
* Standardized the behavior of links to external sites. These now all open in new tabs. No change to the behavior of links that stay on docs.base.org, these continue to open in the current tab.

**How has it been tested?**
Locally. I ensured that all clicks result in events with the correct labels.

### CODE REVIEW

File: apps/base-docs/src/theme/Navbar/Content/index.js [hunk 1]
```diff
@@ -50,6 +51,9 @@ function NavbarLayoutTopContent({ left, right }) {
             target="_blank"
             rel="noreferrer"
             aria-label="Base on Discord"
+            onClick={() => {
+              logEvent('DiscordButton_Clicked', {})
```
Comment (zencephalon) [inline apps/base-docs/src/theme/Navbar/Content/index.js:55]: Let's make these callbacks

Comment (brendan-defi) [reply apps/base-docs/src/theme/Navbar/Content/index.js]: Addressed in the latest.

File: apps/base-docs/src/theme/Navbar/Content/index.js [hunk 2]
```diff
@@ -50,6 +55,9 @@ function NavbarLayoutTopContent({ left, right }) {
             target="_blank"
             rel="noreferrer"
             aria-label="Base on Discord"
+            onClick={() => {
```
Comment (zencephalon) [inline apps/base-docs/src/theme/Navbar/Content/index.js:58]: These can all be callbacks

File: apps/base-docs/docusaurus.config.js [hunk 3]
```diff
@@ -185,6 +185,8 @@ const config = {
           to: '/tutorials',
           navposition: 'bottomLeft',
           label: 'Tutorials',
+          type: 'custom-navbarLink',
+          eventLabel: 'TutorialsButton_Clicked',
```
Comment (zencephalon) [inline apps/base-docs/docusaurus.config.js:189]: Did we want to try to align event naming with the best practices as mentioned in documentation? https://docs.cbhq.net/frontend/analytics/log-events#best-practices

File: apps/base-docs/src/theme/Navbar/Content/index.js (hunk 2 above)
Comment (brendan-defi) [reply apps/base-docs/src/theme/Navbar/Content/index.js]: Understood, I've created callbacks for each social click in the next PR.

File: apps/base-docs/docusaurus.config.js (hunk 3 above)
Comment (brendan-defi) [reply apps/base-docs/docusaurus.config.js:189]: Yes, Gardo is planning to give me updated event names, hopefully today.

File: apps/base-docs/src/theme/Navbar/Content/index.js [hunk 4]
```diff
@@ -39,6 +40,19 @@ ${JSON.stringify(item, null, 2)}`,
   );
 }
 function NavbarLayoutTopContent({ left, right }) {
+  const discordClick = useCallback(() => {
+    logEvent('navbar_social_discord', {});
```
Comment (zencephalon) [inline apps/base-docs/src/theme/Navbar/Content/index.js:44]: Let's make sure these get updated too with event metadata

File: apps/base-docs/src/components/OnchainSummer/OcsBanner.tsx [hunk 5]
```diff
@@ -10,6 +11,10 @@ const href = 'https://www.base.org/onchainsummer?utm_source=DocsSite&utm_campaig
 export function OcsBanner() {
   const [isBannerVisible, setIsBannerVisible] = useLocalStorage('isOcsBannerVisible', true);
 
+  const linkClick = useCallback(() => {
+    logEvent('navbar_ocsbanner', {});
```
Comment (zencephalon) [inline apps/base-docs/src/components/OnchainSummer/OcsBanner.tsx:15]: Let's get eventType info in here too

**Review by zencephalon — DISMISSED** [review]

**Review by zencephalon — APPROVED** [review]

**Pull request merged**